libdnf/repo/RpmPackage.cpp

```diff
@@ -0,0 +1,650 @@
+#include "RpmPackage.hpp"
+#include "libdnf/hy-nevra.hpp"
+#include "libdnf/dnf-advisory-private.hpp"
+#include "libdnf/dnf-packagedelta-private.hpp"
+#include "libdnf/utils/utils.hpp"
+#include "libdnf/hy-util.h"
+#include "libdnf/sack/advisory.hpp"
+
+#include <map>
+#include <solv/evr.h>
+#include <sstream>
+#include <fcntl.h>
+
+class RpmPackage::Impl {
+public:
+    Impl() = default;
+    Impl(const Impl &source);
+private:
+    friend class RpmPackage;
+
+    LrChecksumType transformToLrChecksumType(GChecksumType checksum);
+    std::string buildPackageId(const std::string &name, const std::string &version, const std::string &arch,
+                               const std::string &data) const;
+
+    bool userAction;
+    DnfStateAction action;
+    DnfPackageInfo info;
+    std::shared_ptr<DnfRepo> repo;
+    DnfSack *sack;
+    std::string filename;
+    unsigned char *headerHash;
+    std::string origin;
+    std::string packageId;
+};
```

Comment:
I am not sure if this is a direction that we should follow. The original DnfPackage had only Id, and DnfSack*. Other information were stored in DnfSack/Pool. Why we have to duplicate data or make a Class so huge?

Reply:
Original DnfPackage had more than Id and DnfSack, look properly.

Reply:
I am sorry I forget gboolean loaded. And I am not going to count 8 reserved pointer positions.

Reply:
Once again, look properly: https://github.com/rpm-software-management/libdnf/blob/22480c6480dbe7401b2acba4937d282cab1599ce/libdnf/dnf-package.cpp#L48-L57